eras/conway/impl/src/Cardano/Ledger/Conway/Governance.hs

```diff
@@ -339,7 +349,79 @@ instance EraPParams era => ToCBOR (PulsingSnapshot era) where
 instance EraPParams era => FromCBOR (PulsingSnapshot era) where
   fromCBOR = fromEraCBOR @era
 
--- ===============================
+pgacPParamUpdateL ::
+  Lens' (PrevGovActionIdsChildren era) (Set (GovActionId (EraCrypto era)))
+pgacPParamUpdateL =
+  lens
+    (unPrevGovActionChildren . pgacPParamUpdate)
+    $ \x y -> x {pgacPParamUpdate = PrevGovActionChildren y}
```

Comment:
Is it really okay to make lenses like these that help us make changes easily but bypass the type safety that has been carefully put together for the data-type?

Reply:
Should these lenses be broken into two levels so that they can be used as a composition at their use sites?

Reply:
It is no different from using the actual record field names, they are exported and allow us to modify this type in very much the same way as the lenses. I don't see any problem here

Reply:
Oh, I see what's going on.By doing this we erase the type safety. Then no, this is not good.